### ISSUE TITLE
Add a scaling factor to scale the RoI box head in fastRCNN

### ISSUE THREAD

**Issue by MichalHek** [facebookresearch/detectron2#3428]
## 🚀 Feature
Add a configurable scaling factor - cfg.MODEL.ROI_BOX_HEAD.BBOX_SCALE_FACTOR, that enables scaling the RoI box heads in the Region Proposal Network.

## Motivation & Examples

This feature enables to enlarge the RoI box for tight detections to enable better inference of the following mask detection network.

Motivation derives from cases where long objects (or any object not covered by the RoI box proposals) are cropped because the box proposal is too tight, which can result in nonoptimal instance segmentation in those boxes.

The scale factor will be added as a new property to the defaults config file:
_C.MODEL.ROI_BOX_HEAD.BBOX_SCALE_FACTOR = 1.0

and an extra argument, bbox_s, will be added as an input to the function apply_deltas() in modeling/box_regression.py :

Code illustration:
def apply_deltas(self, deltas, boxes, bbox_s):
...
        pred_w = torch.exp(dw) * widths[:, None] * bbox_s
        pred_h = torch.exp(dh) * heights[:, None] * bbox_s

Visual illustration:
- Current output from PointRend detector:
![image (1)](https://user-images.githubusercontent.com/34838932/131578075-bfe5923a-a38c-4836-b53b-025d3e1c64ea.png)

- output with scaling factor of 1.1  (cfg.MODEL.ROI_BOX_HEAD.BBOX_SCALE_FACTOR = 1.1)
![image (2)](https://user-images.githubusercontent.com/34838932/131578113-895479e6-70e9-4a4f-9e2d-dd286af5bdb0.png)

**Comment by ppwwyyxx** [comment]
This can already be done easily in users' code:
```
class MyROIHeads(StandardROIHeads):
    def forward_with_given_boxes(self, features, instances):
        instances = make_box_larger(instances)
        return super().forward_with_given_boxes(features, instances)
```
